tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/DbScopeFactory.cs

```diff
@@ -100,15 +100,17 @@ internal static class DbScopeFactory
                     }
                     else
                     {
-                        var traceParentInjectedInContext = DatabaseMonitoringPropagator.PropagateDataViaContext(tracer.Settings.DbmPropagationMode, integrationId, command.Connection, scope.Span);
                         var propagatedCommand = DatabaseMonitoringPropagator.PropagateDataViaComment(tracer.Settings.DbmPropagationMode, tracer.DefaultServiceName, tagsFromConnectionString.DbName, tagsFromConnectionString.OutHost, scope.Span, integrationId, out var traceParentInjectedInComment);
                         if (!string.IsNullOrEmpty(propagatedCommand))
                         {
                             command.CommandText = $"{propagatedCommand} {commandText}";
-                            if (traceParentInjectedInComment || traceParentInjectedInContext)
-                            {
-                                tags.DbmTraceInjected = "true";
-                            }
+                        }
+
+                        // try context injection only after comment injection, so that if it fails, we still have service level propagation
+                        var traceParentInjectedInContext = DatabaseMonitoringPropagator.PropagateDataViaContext(tracer.Settings.DbmPropagationMode, integrationId, command.Connection, scope.Span);
```

Comment:
Does it make sense to propagate context if `string.IsNullOrEmpty(propagatedCommand)`?
If yes, then this change is valid (and I wonder why we had it under the condition `!string.IsNullOrEmpty(propagatedCommand)` before.
If no, then I guess we need to find a different way (eg try/catch)?

Reply:
`propagatedCommand` is badly named, it's actually the comment (I'll push a commit to rename it). If the comment is null or empty, we probably won't do anything in the other method, but it's better to check the underlying reasons (type of propagation, type of DB) rather than couple context injection to having a comment.